tokio/src/sync/broadcast.rs

```diff
@@ -961,11 +972,12 @@ impl<T> Slot<T> {
             }
         }
 
-        let prev = self.lock.fetch_sub(2, SeqCst);
-
-        if prev & 1 == 1 {
-            // Sender waiting for lock
-            condvar.notify_one();
+        if 1 == self.lock.fetch_sub(2, SeqCst) - 2 {
+            // First acquire the lock to make sure our sender is waiting on the
+            // condition variable, otherwise the notification could be lost.
+            let _ = tail.lock().unwrap();
```

Comment:
If I understand what is going on, `drop(tail.lock().unwrap())` should be sufficient.

Reply:
`let _ = tail.lock().unwrap();` and `drop(tail.lock().unwrap())` are the same?. Or perhaps I don't understand what you mean.

Reply:
No, you are correct. I don't know why I read it wrong.